pallets/loans/src/functions.rs

```diff
@@ -125,68 +125,99 @@ impl<T: Config> Pallet<T> {
 		Ok(loan_id)
 	}
 
-	pub(crate) fn price_loan(
+	pub(crate) fn price_created_loan(
 		pool_id: PoolIdOf<T>,
 		loan_id: T::LoanId,
 		interest_rate_per_sec: T::Rate,
 		loan_type: LoanType<T::Rate, T::Balance>,
 	) -> Result<u32, DispatchError> {
-		Loan::<T>::try_mutate(pool_id, loan_id, |loan| -> Result<u32, DispatchError> {
-			let loan = loan.as_mut().ok_or(Error::<T>::MissingLoan)?;
+		let now = Self::now();
+		ensure!(loan_type.is_valid(now), Error::<T>::LoanValueInvalid);
 
-			// ensure loan is created or priced but not yet borrowed against
-			ensure!(
-				loan.status == LoanStatus::Created || loan.status == LoanStatus::Active,
-				Error::<T>::LoanIsActive
-			);
-			let mut active_loans = ActiveLoans::<T>::get(pool_id);
-			if loan.status == LoanStatus::Active {
-				if let Some((idx, active_loan)) = active_loans
-					.iter()
-					.enumerate()
-					.find(|(_, active_loan)| active_loan.loan_id == loan_id)
-				{
-					ensure!(
-						active_loan.total_borrowed == Zero::zero(),
-						Error::<T>::LoanIsActive
-					);
-					active_loans.remove(idx);
-				}
-			}
+		ensure!(
+			interest_rate_per_sec >= One::one(),
+			Error::<T>::LoanValueInvalid
+		);
 
-			// ensure loan_type is valid
-			let now = Self::now();
-			ensure!(loan_type.is_valid(now), Error::<T>::LoanValueInvalid);
+		let active_loan = PricedLoanDetails {
+			loan_id,
+			loan_type,
+			interest_rate_per_sec,
+			origination_date: None,
+			normalized_debt: Zero::zero(),
+			total_borrowed: Zero::zero(),
+			total_repaid: Zero::zero(),
+			write_off_status: WriteOffStatus::None,
+			last_updated: now,
```

Comment:
I have changed this line in order to use the same time that was previously checked instead of a call to `Self::now()` to avoid storing a time that has not been checked. I think this should be the expected behaviour, but I would thank a confirmation.